### DISCUSSION TITLE
How to create a closed chain KinematicTree

### DISCUSSION THREAD

**Discussion by ManuelZ** [jgerstmayr/EXUDYN#74] [Q&A on usage of Exudyn]
![Question](https://github.com/jgerstmayr/EXUDYN/assets/115771/91f36b12-472b-4cb9-90b1-4f857fc69ce6)

I think I need to use three `ObjectKinematicTree`s and some `MarkerKinematicTreeRigid`s, but I have some doubts:

a) Can I add three `MarkerKinematicTreeRigid`s to Tree0 and Tree3 and connect them in three points?
    -> Connect Tree0,Body1 with Tree3,Body1 with a revolute joint
    -> Connect Tree0,Body2 with Tree3,Body0 through an `ObjectConnectorSpringDamper`
    -> Connect Tree0,Body2 with Tree3,Body0 through a prismatic joint
b) To create the contact with the ground, can I just use `AddSphereWithMarker`?
c) Does this make sense?

**Comment by jgerstmayr** [comment]
You can combine several ObjectKinematicTree, but it requires constraints and thus requires an implicit solver. An alternative is to connect several  ObjectKinematicTree with spring-dampers, but then you need to find appropriate values.

You can attach a rigid-body marker to ObjectKinematicTree and hereafter use it in GeneralContact.

**Comment by ManuelZ** [comment]
Thank you! That clarifies the matter a lot.

For future reference: I think one starting point on how to select the appropiate stiffness/damping values can be found in this part of the Drake's documentation: [How to choose a force stiffness constant kx or damping constant dx](https://drake.mit.edu/doxygen_cxx/classdrake_1_1multibody_1_1_linear_bushing_roll_pitch_yaw.html#Basic_bushing_force_stiffness_and_damping). There is also information in the following document, in the section ["Estimating bushing parameters"](https://github.com/RobotLocomotion/drake/blob/3befa67f80d4f4c27f7b0f2379776a82e395bc8c/examples/multibody/four_bar/README.md?plain=1#L129).
